CryptoPkg/Library/BaseCryptLibMbedTls/Pk/CryptRsaPssSign.c

```diff
@@ -0,0 +1,231 @@
+/** @file
+  RSA PSS Asymmetric Cipher Wrapper Implementation over MbedTLS.
+
+  This file implements following APIs which provide basic capabilities for RSA:
+  1) RsaPssSign
+
+Copyright (c) 2023, Intel Corporation. All rights reserved.<BR>
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include "InternalCryptLib.h"
+#include <mbedtls/rsa.h>
+#include <Library/RngLib.h>
+
+/**
+  The MbedTLS function f_rng, which MbedRand implements, is not
+  documented well.
+
+  @param[in]       RngState  RngState.
+  @param[in]       Output    Output.
+  @param[in]       Len       Len.
+
+  @retval  0                 success.
+  @retval  non-zero          failed.
+
+**/
+INT32
+MbedRand (
+  VOID   *RngState,
+  UINT8  *Output,
+  UINTN  Len
+  )
+{
+  BOOLEAN  Ret;
+  UINT64   TempRand;
+
+  Ret = FALSE;
+
+  while (Len > 0) {
+    // Use RngLib to get random number
+    Ret = GetRandomNumber64 (&TempRand);

```

Comment:
not to be dependent on the internal of the GetRandomNumber64 for zeroizing of TempRand we may need to zeroize it for every iteration that way we don't get any duplicate values.

Reply:
Thanks. I have initialized `TempRand` to zero.